### PR TITLE
Update composer dependencies

### PR DESCRIPTION
**Pull request by BorisovskiP**
Update required dependencies so they are compatible with Magento 2.4

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
+        "magento/module-catalog": "~103.0||104.0",
         "magento/module-store": "~101.0",
         "magento/module-ui": "~101.1",
-        "php": "~7.1.3||~7.2.0||~7.3.0"
+        "php": "~7.1.3||~7.2.0||~7.3.0||7.4.0"
     },
     "type": "magento2-module",
     "license": "MIT",
-    "version": "1.1.2",
+    "version": "1.1.3",
```
Comment (sprankhub) [inline composer.json]: @Spaggel, I would suggest removing the version from the `composer.json` as this leads to issues. E.g. see https://blog.packagist.com/tagged-a-new-release-for-composer-and-it-wont-show-up-on-packagist/. Do you agree?

**Review by sprankhub — CHANGES_REQUESTED** [review]
Multiple templates and a function in the JS mixin are overridden from the core. If we establish compatibility with Magento 2.4, these should be checked and updated IMHO.

File: composer.json (hunk 1 above)
Comment (Spaggel) [reply composer.json]: Agreed!

**Comment by Spaggel** [comment]
@BorisovskiP thanks for your pull request. I'm currently unable to test the changes myself. Are you sure everything keeps working after your change?

**Comment by BorisovskiP** [comment]
@Spaggel i would say that the extension should work as normal as we didn't made any major changes to it.

**Comment by sprankhub** [comment]
I am pretty sure these changes will break 2.3 compatibility. Hence, I suggest:

1. Remove old version constraints from `composer.json` and only use the 2.4 ones.
2. State that this extension is compatible with Magento 2.3 until version `1.1.2` only.
2. Release new version as `1.2.0` or even `2.0.0`.

**Comment by BorisovskiP** [comment]
@Spaggel Could you review the updated changes made to the PR? 

What do you think about @sprankhub suggestions?

**Comment by Spaggel** [comment]
Hi @BorisovskiP, I'm not quite sure if we're ready to approve; it depends on what Simon has to say.
@sprankhub should the ` "magento/framework": "~102.0||~103.0",` be replaced with just `~103.0` (for all magento/* dependencies) to target 2.4 instead of 2.3 and 2.4?
If so, we should add a note to the readme about the version compatibility (like Simon suggested).
I will release it as version `1.2.0` afterward.

Thanks for your contribution @BorisovskiP and your time @sprankhub 😄

File: composer.json [hunk 2]
```diff
@@ -2,16 +2,15 @@
...
-        "magento/framework": "~102.0",
-        "magento/module-backend": "~101.0",
-        "magento/module-catalog": "~103.0",
+        "magento/framework": "~103.0",
+        "magento/module-backend": "~102.1",
+        "magento/module-catalog": "~104.0",
         "magento/module-store": "~101.0",
         "magento/module-ui": "~101.1",
-        "php": "~7.1.3||~7.2.0||~7.3.0"
+        "php": "~7.3.0||7.4.0"
```
Comment (sprankhub) [inline composer.json]: This should be `~7.3.0||~7.4.0`.

File: composer.json [hunk 3]
```diff
@@ -2,16 +2,15 @@
     "name": "spaggel/tooltip",
     "description": "Product attribute tooltips module for Magento 2",
     "require": {
-        "magento/framework": "~102.0",
-        "magento/module-backend": "~101.0",
-        "magento/module-catalog": "~103.0",
+        "magento/framework": "~103.0",
+        "magento/module-backend": "~102.1",
```
Comment (sprankhub) [inline composer.json]: Where did you get these dependencies from? Magento 2.4 has `magento/module-backend` in version `102.0.1`. Hence, by doing this, we only support `2.4.1`. Even though people should use `2.4.1` already, I do not see the point of restricting this so much.

**Review by sprankhub — CHANGES_REQUESTED** [review]

**Review by sprankhub — APPROVED** [review]

**Comment by sprankhub** [comment]
Looks fine to me. What do you think, @Spaggel?

**Comment by Spaggel** [comment]
Looks good to me!

**Pull request merged**